src/lib/dhcp/tests/option6_auth_unittest.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2011-2015 Internet Systems Consortium, Inc. ("ISC")
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <dhcp/dhcp6.h>
+#include <dhcp/option.h>
+#include <dhcp/option6_auth.h>
+#include <util/buffer.h>
+
+#include <gtest/gtest.h>
+#include <boost/scoped_ptr.hpp>
+
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+using namespace isc;
+using namespace isc::dhcp;
+using namespace isc::util;
+using boost::scoped_ptr;
+
+namespace {
+class Option6AuthTest : public ::testing::Test {
+public:
+    Option6AuthTest(): buff_(28) { 
+    }
+    OptionBuffer buff_;
+};
+
+// check constructor, setters and getters 
+TEST_F(Option6AuthTest, basic) { 
+
+    scoped_ptr<Option6Auth> auth; 
+    ASSERT_NO_THROW(auth.reset(new Option6Auth(1,2,0,0x9000,{'a','b','c','d'})));
+
+    ASSERT_EQ(1, auth->getProtocol());
+    ASSERT_EQ(2, auth->getHashAlgo());
+    ASSERT_EQ(0, auth->getRplyDtctnMthd());
+    ASSERT_EQ(0x9000, auth->getRplyDtctnValue());
+    
+    std::vector<uint8_t> test_buf = {'a','b','c','d'};
+    ASSERT_EQ(test_buf, auth->getAuthInfo());
+
+    auth->setProtocol(2);
+    auth->setHashAlgo(3);
+    auth->setRplyDtctnMthd(1);
+    auth->setRplyDtctnValue(109034830);
+    auth->setAuthInfo({1,2,3,4});
+
+    ASSERT_EQ(2, auth->getProtocol());
+    ASSERT_EQ(3, auth->getHashAlgo());
+    ASSERT_EQ(1, auth->getRplyDtctnMthd());
+    ASSERT_EQ(109034830, auth->getRplyDtctnValue());
+    
+    test_buf = {1,2,3,4};
+    ASSERT_EQ(test_buf, auth->getAuthInfo());
+}
+
+//Check if all the fields are properly parsed and stored
+// todo define userdefined literal and add packing function to it
+TEST_F(Option6AuthTest, parseFields) { 
+    buff_[0] = 0xa1; //protocol
+    buff_[1] = 0xa2; //algo
+    buff_[2] = 0xa3; //rdm method
+    buff_[3] = 0xa4; //rdm value
+    buff_[4] = 0xa5; //rdm value
+    buff_[5] = 0xa6; //rdm value
+    buff_[6] = 0xa7; //rdm value
+    buff_[7] = 0xa8; //rdm value
+    buff_[8] = 0xa9; //rdm value
+    buff_[9] = 0xaa; //rdm value
+    buff_[10] = 0xab; //rdm value
+    for ( uint8_t i = 11; i < 27; i++ ) {
+        buff_[i] = 0xa8; //auth info 16 bytes
```

Comment:
It's better to use different values for each byte. If the whole 16 bytes are the same, how do you know if the code really read each of them once and not for example the first 4 bytes were read 4 times?